go/extractor/extractor.go

```diff
@@ -89,6 +89,7 @@ func ExtractWithFlags(buildFlags []string, patterns []string) error {
 			packages.NeedTypes | packages.NeedTypesSizes |
 			packages.NeedTypesInfo | packages.NeedSyntax,
 		BuildFlags: buildFlags,
+		Tests:      extractTests,
```

Comment:
It may be worth adding a short comment here to note that this may instruct the Go compiler to build tests in autobuild mode if `CODEQL_EXTRACTOR_GO_EXTRACT_TESTS` is `true`.

Reply:
Perhaps we could also modify the log message above (on L84) to include something to note if tests are being extracted. Might be helpful for troubleshooting purposes.

Reply:
Done; I'll also have this message note vendor extraction status once the other PR is merged.